clang/lib/DPCT/UserDefinedRules/PatternRewriter.cpp

```diff
@@ -295,24 +324,36 @@ static int parseCodeElement(const MatchPattern &Suffix,
         return Index;
       }
 
-      if (isRightDelimiter(Character) || Index == Size - 1) {
-        return -1;
+      if (isPythonScript) {
+        // For parsing python syntax inside braces ([[a.cu]]), don't return
+        // immediately when the current character is a right delimiter (']').
+        // Continue parsing for non right delimiter suffix literals ('.cu').
+        if ((isRightDelimiter(suffixLit) && isRightDelimiter(Character)) ||
+            Index == Size - 1) {
+          return -1;
+        }
+      } else {
+        if (isRightDelimiter(Character) || Index == Size - 1) {
+          return -1;
+        }
       }
     }
 
-    if (Character == '{') {
-      Index = parseBlock('{', '}', Input, Index, Mode);
-      continue;
-    }
+    if (isLeftDelimiter(Character)) {
+      char rightDelimiter = getRightDelimiter(Character);
 
-    if (Character == '[') {
-      Index = parseBlock('[', ']', Input, Index, Mode);
-      continue;
-    }
-
-    if (Character == '(') {
-      Index = parseBlock('(', ')', Input, Index, Mode);
-      continue;
```

Comment:
And I think to parse the content inside, `Index` should be `++Index`

Reply:
Index is getting incremented inside parseBlock [here](https://github.com/oneapi-src/SYCLomatic/blob/SYCLomatic/clang/lib/DPCT/UserDefinedRules/PatternRewriter.cpp#L253)

Did you mean the same or more additional increment?

Reply:
>> Index is getting incremented inside parseBlock [here](https://github.com/oneapi-src/SYCLomatic/blob/SYCLomatic/clang/lib/DPCT/UserDefinedRules/PatternRewriter.cpp#L253)
You are right.